File: fixtures/dart/put_xput.dart
import 'package:http/http.dart' as http;

void main() async {
  var headers = {
    'Content-Type': 'application/json',
  };

  var params = {
    'pretty': '',
  };
  var query = params.entries.map((p) => '${p.key}=${p.value}').join('&');

  var data = '{"properties": {"email": {"type": "keyword"}}}';

  var res = await http.put('http://localhost:9200/twitter/_mapping/user?$query', headers: headers, body: data);
  if (res.statusCode != 200) throw Exception('http.put error: statusCode= ${res.statusCode}');
  print(res.body);
}

File: fixtures/python/put_xput.py
import requests

headers = {
    'Content-Type': 'application/json',
}

params = [
    ('pretty', ''),
]

json_data = {
    'properties': {
        'email': {
            'type': 'keyword',
        },
    },
}

response = requests.put('http://localhost:9200/twitter/_mapping/user', headers=headers, params=params, json=json_data)

# Note: the data is posted as JSON, which might not be serialized
# by Requests exactly as it appears in the original command. So
# the original data is also given.
#data = '{"properties": {"email": {"type": "keyword"}}}'
#response = requests.put('http://localhost:9200/twitter/_mapping/user', headers=headers, params=params, data=data)
